toplevel/coqargs.ml

```diff
@@ -464,12 +464,8 @@ let parse_args ~help ~init arglist : t * string list =
 
     |"-w" | "-W" ->
       let w = next () in
-      if w = "none" then
-        (CWarnings.set_flags w; oval)
-      else
-        let w = CWarnings.get_flags () ^ "," ^ w in
-        CWarnings.set_flags (CWarnings.normalize_flags_string w);
-        oval
+      if w = "none" then add_set_option oval ["Warnings"] (Stm.OptionSet(Some w))
+      else add_set_option oval ["Warnings"] (Stm.OptionAppend w)
```

Comment:
This special casing is definitely useless if the two calls end up doing the exact same thing (see my previous comment / question).